pkg/odo/cli/logs/logs.go

```diff
@@ -123,34 +122,41 @@ func (o *LogsOptions) Run(ctx context.Context) error {
 	default:
 		mode = odolabels.ComponentAnyMode
 	}
-	containersLogs, err = o.clientset.LogsClient.GetLogsForMode(mode, o.componentName, o.Context.GetProject(), o.follow)
-	if err != nil {
-		return err
-	}
-	if len(containersLogs) == 0 {
-		// This will be the case when:
-		// 1. user specifies --dev flag, but the component's running in Deploy mode
-		// 2. user specified --deploy flag, but the component's running in Dev mode
-		// 3. user passes no flag, but component is running in neither Dev nor Deploy mode
-		fmt.Fprintf(o.out, "no containers running in the specified mode for the component %q\n", o.componentName)
-		return nil
-	}
+
+	errChan := make(chan error) // errors are put on this channel
+	done := make(chan struct{}) // a struct{}{} on done channel indicates that logs for all pods are fetched
+
+	go func() {
```

Comment:
The content of GetLogsForMode is already executed in g a go routine. Do you really need another one here?

Reply:
This go routine has been removed. We now return a struct of channels from the business layer akin to what you recommended in https://github.com/redhat-developer/odo/pull/5942#issuecomment-1183394909.